src/riscv/lib/src/machine_state/memory/state.rs

```diff
@@ -17,6 +21,14 @@ use crate::state_backend::ManagerRead;
 use crate::state_backend::ManagerReadWrite;
 use crate::state_backend::ManagerWrite;
 
+fn get_page_range(address: Address, length: usize) -> Range<usize> {
+    let address = address as usize;
+    let start_page = address >> super::OFFSET_BITS;
+    let end_page =
+        (address.wrapping_add(length).saturating_sub(1) >> super::OFFSET_BITS).saturating_add(1);
```

Comment:
There is an off-by-one error when `length == 0`. 

When length is 0:
- start address 4095 results in `0..1` (incorrect)
- start address 4096 results in `1..1` (correct)
- start address 4097 results in `1..2` (incorrect)